service/controller/app/v1/resource/clients/resource.go

```diff
@@ -66,6 +78,10 @@ func (r *Resource) addClientsToContext(ctx context.Context, cr v1alpha1.App) err
 		return nil
 	}
 
+	if cc.G8sClient != nil && cc.K8sClient != nil {
```

Comment:
Return early rather than propagate two indentations blocks.

Reply:
Why should the clients be set at this point here? Is the resource called multiple times within the same reconciliation loop? That should not be the case and whenever the resource is called it should mutate the context IMO. Anything else would be confusing and complex.

Reply:
That's right, I should not check `nil` at this point, Thanks to pointing out.